### PR TITLE
US113669 Add turnitin hypermedia

### PR DESCRIPTION
**Pull request by ppaskaris-d2l**
[US113669](https://rally1.rallydev.com/#/detail/userstory/368261329852?fdp=true)

This PR adds turnitin feature status hypermedia, converts it to mobx, and inspects the dialog result to see what they did there. It also does the language/summary changes in the story.

Screenshots
- [Editor](https://user-images.githubusercontent.com/30184468/76653078-5f54eb80-653e-11ea-9f2e-73dc90d98d0f.png)
- [Summary](https://user-images.githubusercontent.com/30184468/76653107-70056180-653e-11ea-8eb1-ec92227c0f04.png)

Dependencies:

- [siren-sdk →](https://github.com/BrightspaceHypermediaComponents/siren-sdk/pull/156)
- [monolith →](https://git.dev.d2l/projects/CORE/repos/lms/pull-requests/9440/overview)

### CODE REVIEW

File: test/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment.spec.js [hunk 1]
```diff
@@ -26,6 +26,10 @@ describe('Assignment ', function() {
 				canEditName: () => true,
 				instructionsEditorHtml: () => 'These are your instructions',
 				canEditInstructions: () => true,
+				canEditTurnitin: () => undefined,
```
Comment (pureooze) [inline test/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment.spec.js]: Does the store actually have these as undefined? Seems like some of these are true/false.

Comment (ppaskaris-d2l) [reply test/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment.spec.js]: The store will use whatever this returns, yup. This is just some mock for unrelated tests, not super concerned with what values it uses.

File: components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment.js [hunk 2]
```diff
@@ -91,6 +95,11 @@ export class Assignment {
 		this._setValidCompletionTypeForSubmissionType();
 	}
 
+	setTurnitin(isOriginalityCheckEnabled, isGradeMarkEnabled) {
```
Comment (pureooze) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/state/assignment.js]: Would be nice to add a test for this 😃

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-assignment-turnitin-editor.js [hunk 3]
```diff
@@ -85,33 +87,93 @@ class AssignmentTurnitinEditor extends EntityMixinLit(LocalizeMixin(LitElement))
...
+			isGradeMarkEnabled
+		} = entity;
+
+		const isTurnitinEnabled = isOriginalityCheckEnabled || isGradeMarkEnabled;
+		const isEditorDisplayed = canEditTurnitin || isTurnitinEnabled;
+
+		let featureSummary;
+		if (isTurnitinEnabled) {
+
+			let originalityCheckItem;
```
Comment (pureooze) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-assignment-turnitin-editor.js]: nit: Maybe these could be their own methods?

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-assignment-turnitin-editor.js [hunk 4]
```diff
@@ -85,33 +87,93 @@ class AssignmentTurnitinEditor extends EntityMixinLit(LocalizeMixin(LitElement))
...
+			/*      responseDataKey: */ 'result',
+			/*                width: */ 960,
+			/*               height: */ 960,
 			/*            closeText: */ this.localize('btnCloseDialog'),
 			/*              buttons: */ buttons,
 			/* forceTriggerOnCancel: */ false
 		);
+		delayedResult.AddListener(result => {
+
+			const resultIsValid = Array.isArray(result) && result.length >= 2;
```
Comment (pureooze) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-assignment-turnitin-editor.js]: Why does result length need to be >= 2?

Comment (ppaskaris-d2l) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-assignment-turnitin-editor.js]: To protect the destructuring on the next line, e.g. if this code is present without the monolith changes.

**Pull request merged**